### ISSUE TITLE
Version-numbered packages are always re-cloned

### ISSUE THREAD

**Issue by rdeits** [rened/DeclarativePackages.jl#6]
I've just started playing with DeclarativePackages, and it looks like an excellent tool. However, I've so far had issues with it performing a lot of seemingly unnecessary package clones. It seems that the hardlinking logic never fires for any package which is listed with a version number instead of a git sha. So, for example, if I have a DECLARE file with:

```
StatsBase 0.8.0
StatsFuns 0.2.0
URIParser 0.1.3
WoodburyMatrices 0.1.5
ZMQ 0.3.1
```

then _any_ change to the DECLARE will result in a new copy StatesBase, StatsFuns, etc. That means that all of those packages will also have to be re-built and re-compiled by Julia. That means that any change to any package results in ~10 minutes of building and precompiling (mostly for libraries like Cairo and Gtk). 

So, first of all, is this expected? Should two packages with the same version number trigger the hardlinking logic? 

My suspicion is that this is actually a bug, since `installpackages.jl` defines what looks like a useful function `metadatacommit()` to extract a git sha from a version number, but that function is never called: https://github.com/rened/DeclarativePackages.jl/blob/master/src/installpackages.jl#L194

I'm happy to try to resolve this myself, if it is a bug.

**Comment by rdeits** [comment]
With https://github.com/rdeits/DeclarativePackages.jl/commit/ac18e1fcca23cf85ec124efb45e77439ece4a23c I can avoid re-cloning unchanged version-numbered packages. They still get pre-compiled, unfortunately, but I'm not sure how to avoid that yet.

**Comment by rened** [comment]
Fixed via 9fe6cbb, also the re-building should be fixed now.
